src/sentry/sentry_metrics/querying/visitors/query_expression.py

```diff
@@ -406,8 +402,30 @@ class NumericScalarsNormalizationVisitor(QueryExpressionVisitor[QueryExpression]
     def __init__(self, unit: Unit):
         self._unit = unit
 
+    def _visit_formula(self, formula: Formula) -> TVisited:
+        has_coefficient_operators = formula.function_name in COEFFICIENT_OPERATORS
+
+        # In case the formula has a coefficient operator with all scalars, we want to scale the entire formula by
+        # wrapping it in another formula. For all the other cases, we just want to apply the scaling to each component
+        # of the formula, to make the formula less deep.
+        # Example:
+        #  scaling (a * b) by 1000 = (a * b) * 1000
+        #  scaling (a + b) by 1000 = (a * 1000 + b * 1000) in this case the multiplication is performed in-memory
+        if has_coefficient_operators:
+            has_all_scalars = True
+            for parameter in formula.parameters:
+                if not self._is_numeric_scalar(parameter):
+                    has_all_scalars = False
+
+            return self._unit.apply_on_query_expression(formula) if has_all_scalars else formula
```

Comment:
Another approach here would be to just apply the scalar to an arbitrary element of the scalar parameters, but this seemed a bit cleaner.